### PR TITLE
Add extensions for `OccurrenceConstraint` that reads more fluently

### LINKED ISSUE #2045: Include extension methods for `OccurrenceConstraint` that reads more fluently
I was wondering if it would make sense to include an extension method for `Exactly.Times` like this:

```cs
myString.Should().Contain("something", 4.Times()); // Equivalent to Exactly.Times(4)
```

I think it reads a bit better than the existing `Exactly.Times(4)`, but perhaps this has already been considered?

I understand there might be some confusion around `AtLeast`, `AtMost` etc., but that might be fixed like:
```cs
myString.Should().Contain("something", 4.TimesOrMore()); // Equivalent to AtLeast.Times(4) and MoreThan(3)
myString.Should().Contain("something", 4.TimesOrLess()); // Equivalent to AtMost.Times(4) and LessThan(5)
```

There seems to be some redundancy in with `AtLeast`/`MoreThan` and `AtMost`/`LessThan`, but they might be useful in scenarios I've never had.

I know creating such extensions is pretty straightforward:
```cs
public static class FluentOccurrenceConstraintExtensions
{
    public static OccurrenceConstraint Times(this int times)
    {
        return Exactly.Times(times);
    }

    public static OccurrenceConstraint TimesOrLess(this int times)
    {
        return AtLeast.Times(times);
    }

    public static OccurrenceConstraint TimesOrMore(this int times)
    {
        return AtMost.Times(times);
    }
}
```

_Originally posted by @siewers in https://github.com/fluentassertions/fluentassertions/discussions/2001_

### PR DESCRIPTION
**Pull request by IT-VBFK**
Authored by @siewers, committed by @IT-VBFK

Closes #2045 

## IMPORTANT 

* [x] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [x] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [x] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [x] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [ ] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).

### CODE REVIEW

**Comment by dennisdoomen** [comment]
Is this still work in progress?

**Comment by IT-VBFK** [comment]
Not really, why?

**Comment by dennisdoomen** [comment]
Just checking whether I can start reviewing it.

**Comment by IT-VBFK** [comment]
Oh.. yes :)
sorry

**Comment by jnyrup** [comment]
Please, can we agree on the API in #2045 first. :pray: 

I see I really need to get this written, but the label "api-suggestion" is supposed to mean "this is a suggestion for a modification to the public api, but it is not yeat ready for implementation". 
"api-approved" is "this is ready for implementation".

**Comment by IT-VBFK** [comment]
> Please, can we agree on the API in https://github.com/fluentassertions/fluentassertions/issues/2045 first. 🙏

Then.. I will make this a WIP..

File: docs/_pages/releases.md [hunk 1]
```diff
@@ -7,8 +7,6 @@ sidebar:
   nav: "sidebar"
 ---
 
-## Unreleased
-
```
Comment (jnyrup) [inline docs/_pages/releases.md:11]: This should be kept.

Comment (IT-VBFK) [reply docs/_pages/releases.md]: Oh.. accidentally removed

File: Tests/FluentAssertions.Specs/OccurrenceConstraintSpecs.cs [hunk 2]
```diff
@@ -25,17 +28,24 @@ public class OccurrenceConstraintSpecs
...
         { LessThan.Thrice(), 2 },
         { LessThan.Times(4), 3 },
+        { 4.TimesOrLess(), 4 },
+        { 4.TimesOrLess(), 1 },
         { MoreThan.Once(), 2 },
         { MoreThan.Twice(), 3 },
         { MoreThan.Thrice(), 4 },
         { MoreThan.Times(4), 5 },
+        { 4.TimesOrMore(), 4 },
+        { 4.TimesOrMore(), 5 },
```
Comment (jnyrup) [inline Tests/FluentAssertions.Specs/OccurrenceConstraintSpecs.cs:48]: These cases are already included above.
Same for `FailingConstraints`.

**Review by jnyrup — APPROVED** [review]

File: Tests/FluentAssertions.Specs/OccurrenceConstraintSpecs.cs (hunk 2 above)
Comment (IT-VBFK) [reply Tests/FluentAssertions.Specs/OccurrenceConstraintSpecs.cs]: Thanks for the hint.

**Comment by IT-VBFK** [comment]
Just a note: Please make sure to add the line "Authored by @siewers, committed by @IT-VBFK" in the commit message, to give @siewers the credit for the change, because it is actually his work :)

**Review by dennisdoomen — CHANGES_REQUESTED** [review]
Please also update `strings.md` to show the new way.

File: docs/_pages/strings.md [hunk 3]
```diff
@@ -115,3 +115,11 @@ And if that's not enough, you can assert on the number of matches of a regular e
 someString.Should().MatchRegex("h.*\\sworld.$", Exactly.Once());
 someString.Should().MatchRegex(new Regex("h.*\\sworld.$"), AtLeast.Twice());
 ```
+
+If you have feelings that `Exactly.Times(4)`, `AtLeast.Times(4)` and `AtMost.Times(4)` reads a bit odd or bumpy, you now can do following:
```
Comment (jnyrup) [inline docs/_pages/strings.md:119]:
```suggestion
If you prefer a more fluent syntax than `Exactly.Times(4)`, `AtLeast.Times(4)` and `AtMost.Times(4)` reads, you can do the following:
```

File: Src/FluentAssertions/Extensions/OccurrenceConstraintExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
+using System;
+
+namespace FluentAssertions.Extensions;
+
+
+/// <summary>
+/// Provides little extensions to write less bumpy <see cref="OccurrenceConstraint" />s
```
Comment (jnyrup) [inline Src/FluentAssertions/Extensions/OccurrenceConstraintExtensions.cs:7]:
```suggestion
/// Provides extensions to write <see cref="OccurrenceConstraint" />s with fluent syntax
```

**Review by dennisdoomen — APPROVED** [review]

**Pull request merged**